Task7

VadymSalymonenko opened the pull request:
added code and Cmake for task 7.

VadymT95 committed:
added task1 code and Cmake file.

VadymT95 committed:
added code and Cmake for task 2.

VadymT95 committed:
added code and Cmake for task 3.

VadymT95 committed:
added code and Cmake for task 4.

VadymT95 committed:
added code and Cmake for task 6.

VadymT95 committed:
added code and Cmake for task 7.

BBuinich commented on 7/src/part1.c:
```diff
@@ -0,0 +1,175 @@
+    char name[256];
+    int is_directory;                            // 1 if directory, 0 if file
+    int size;                                    // size of the directory or file
+    struct file_or_dir* contents[MAX_CONTENTS];  // children if it's a directory
```
don't need to allocate memory here, allocate it i  create_directory function

BBuinich commented on 7/src/part1.c:
```diff
@@ -0,0 +1,175 @@
+    new_dir->is_directory = 1;
+    new_dir->size = 0;
+    new_dir->parent = NULL;
+    for (int i = 0; i < MAX_CONTENTS; i++) {
+        new_dir->contents[i] = NULL;
+    }
```
```suggestion
    new_dir->contents = malloc(sizeof(file_or_dir*) * 10);  // Initially allocate space for 10 items
    new_dir->content_count = 0;
```

BBuinich commented on 7/src/part1.c:
```diff
@@ -0,0 +1,175 @@
+
+
+void add_to_directory(file_or_dir* parent, file_or_dir* child) {
+    for (int i = 0; i < MAX_CONTENTS; i++) {
```
why do we need loop here?
why we can do 
    parent->contents[parent->content_count++] = child;
    child->parent = parent; 
?

BBuinich commented on 7/src/part1.c:
```diff
@@ -0,0 +1,175 @@
+}
+
+
+void add_to_directory(file_or_dir* parent, file_or_dir* child) {
```
add realloc of memory here 
This reduces the memory overhead and makes the code more scalable.
    if(parent->content_count % 10 == 0) {
        parent->contents = realloc(parent->contents, sizeof(file_or_dir*) * (parent->content_count + 10));
    }

BBuinich reviewed (changes requested):
perfect!!! leave here only task 7

VadymT95 committed:
Minor changes.

1) added new fild to struct "file_or_dir": "contents";
2) deleted some loops;
3) added realloc to dir->contents.
